TrafficCapture/trafficReplayer/src/testFixtures/java/org/opensearch/migrations/replay/TestUtils.java

```diff
@@ -118,7 +118,7 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest msg) thro
     private static String getStringFromContent(FullHttpRequest fullRequest) throws IOException {
         try (var baos = new ByteArrayOutputStream()) {
             var bb = fullRequest.content();
-            bb.readBytes(baos, bb.readableBytes());
+            bb.duplicate().readBytes(baos, bb.readableBytes());
             return baos.toString(StandardCharsets.UTF_8);
```

Comment:
why not just call .toString(StandardCharsets.UTF_8)?

Reply:
Good catch. Thanks!